### PR TITLE
Remove React 17 requirement for `AfterActivationSiteScan` service 

### LINKED ISSUE #7489: Remove React 17 requirement for certain features
### Feature Description

Let's undo aspects of #7421 which made certain services dependent on React 17. As seen in the below [Make/Core post](https://make.wordpress.org/core/2023/03/07/upgrading-to-react-18-and-common-pitfalls-of-concurrent-mode/#:~:text=New%20API%20for%20mounting%20a%20root), any React 17 code will continue to work with React 18 by invoking a legacy API:

> image

In particular, this is considering the following line (at least):

https://github.com/ampproject/amp-wp/blob/f8249c1506f7d2f0cc59b27f73294cf148fea01b/src/Admin/AfterActivationSiteScan.php#L104

Related: https://github.com/ampproject/amp-wp/pull/7394

### Acceptance Criteria

_No response_

### Implementation Brief

_No response_

### QA Testing Instructions

_No response_

### Demo

_No response_

### Changelog Entry

_No response_

### PR DESCRIPTION
**Pull request by thelovekesh**
## Summary 

Fixes #7489

## Checklist

- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: bin/local-env/start.sh [hunk 1]
```diff
@@ -23,9 +23,6 @@ cd "$(dirname "$0")/../.."
 # done on every time `npm run test-e2e` is run.
 . "$(dirname "$0")/install-wordpress.sh"
 
-# Install specific version of Gutenberg plugin for e2e tests.
-download_gutenberg "14.7.3" "tests/e2e/plugins/gutenberg.14.7.3.zip"
```
Comment (thelovekesh) [inline bin/local-env/start.sh:27]: Just in case we ever need it, should we leave the `download_gutenberg` util function in place?

Comment (thelovekesh) [reply bin/local-env/start.sh:27]: Same for `uploads.ini`

https://github.com/ampproject/amp-wp/blob/96cdfb90e4124ccfc5f99c46d006a8a564479b4f/bin/local-env/docker-compose.yml#L13

**Comment by thelovekesh** [comment]
Looks like flakiness in e2e is still there, even after #7488.

File: bin/local-env/start.sh (hunk 1 above)
Comment (westonruter) [reply bin/local-env/start.sh:27]: We could, but probably better to remove it. It can be resurrected via version control if needed.

**Review by westonruter — APPROVED** [review]
LGTM

**Comment by pavanpatil1** [comment]
QA Passed ✅ 

Verified the issue, the site scan function is working fine with the React 18 and WP6.2 current RC version. 

image

**Pull request merged**